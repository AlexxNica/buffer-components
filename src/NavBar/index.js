import React, { PropTypes } from 'react';
import reactCSS from 'reactcss';

const NavBar = ({ children }) => {
  const styles = reactCSS({
    default: {
      NavBar: {
        alignItems: 'center',
        backgroundColor: '#fff',
        borderBottom: '1px solid #ced7df',
        boxSizing: 'border-box',
        display: 'flex',
        flexDirection: 'row',
        height: 64,
        lineHeight: 64,
        marginBottom: 32,
        padding: '0 16px',
        width: '100%',
      },
    },
  });
  return (
    <div
      style={styles.NavBar}
    >
      {children}
    </div>
  );
};

NavBar.propTypes = {
  children: PropTypes.node,
};

export default NavBar;
