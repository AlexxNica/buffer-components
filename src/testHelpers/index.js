export const renderAndCheckSnapshot = (component) => {
  const renderedComponent = renderer.create(component);
  const tree = renderedComponent.toJSON();
  expect(tree).toMatchSnapshot();
};
