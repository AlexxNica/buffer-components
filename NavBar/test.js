jest.unmock('./index');
jest.unmock('../Icon');
jest.unmock('../Text');
jest.unmock('../VectorGraphic');
import React from 'react';
import { renderAndCheckSnapshot } from '../testHelpers';
import NavBar from './index';

describe('NavBar', () => {
  it('NavBar component', () => {
    renderAndCheckSnapshot(<NavBar />);
  });

  it('NavBar w/title', () => {
    renderAndCheckSnapshot(<NavBar title={'Title'} />);
  });

  it('NavBar w/subtitle', () => {
    renderAndCheckSnapshot(<NavBar subtitle={'subtitle'} />);
  });
});
