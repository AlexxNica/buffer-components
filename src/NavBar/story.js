import React from 'react';
import { storiesOf } from '@kadira/storybook';
import NavBar from './index';
import Image from '../Image';

storiesOf('NavBar')
  .add('Default', () => (
    <NavBar>
      <Image
        src={'https://buffer.com/images/app/buffer-logo@2x.png'}
        height={"15px"}
      />
    </NavBar>
  ));
