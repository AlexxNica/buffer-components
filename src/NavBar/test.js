jest.unmock('./index');
import React from 'react';
import { renderAndCheckSnapshot } from '../testHelpers';
import NavBar from './index';

describe('NavBar', () => {
  it('NavBar component', () => {
    renderAndCheckSnapshot(<NavBar>Navigation Elements</NavBar>);
  });
});
