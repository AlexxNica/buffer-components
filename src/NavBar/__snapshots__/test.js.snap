exports[`NavBar NavBar component 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderBottom": "1px solid #ced7df",
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "row",
      "height": 64,
      "lineHeight": 64,
      "marginBottom": 32,
      "padding": "0 16px",
      "width": "100%"
    }
  }>
  Navigation Elements
</div>
`;
